example-apps/simple/application.yml

```diff
@@ -3,9 +3,15 @@ description: For making testing fast
 author: exospheredev
 version: '1.0'
 
-dependencies:
-  - name: exocom
-    version: 0.26.1
+development:
+  dependencies:
+    - name: exocom
+      version: 0.26.1
+
+production:
+  dependencies:
+    - name: exocom
+      version: 0.26.1
```

Comment:
Rather than duplicating information here, can we add an application-wide `dependencies` block, and have the environments only provide overrides?

```yml
dependencies:
  - name: exocom
    version: 0.26.1

development:
  - name: exocom
    version: 0.27   # new exocom version in dev

production:
  # no overrides here, uses exocom 0.26.1
```

Reply:
We considered that but ultimately decided that it was only a source of confusion and didn't provide that much convenience. See: https://github.com/Originate/exosphere/issues/559#issuecomment-329852538

Reply:
I don't like the thought of merging arrays. Merging objects is much simpler and easier to understand, so if we really want to reduce duplication (and this is not much currently), I think we need to make the dependencies an object.

Reply:
I think a dependency's `config` will likely have to change between environments, in which case there will be one entry per dependency per environment containing at least `name` and `config`. At that point having a generic `dependencies` block might actually be more lines and doesn't make things any clearer.